src/Query/Filter/TextField.ts

```diff
@@ -44,17 +43,24 @@ export abstract class TextField extends Field {
         // Finally, we can create the Filter, that takes a task
         // and tests if it matches the string filtering rule
         // represented by this object.
-        return FilterOrErrorMessage.fromFilter((task: Task) => {
-            return TextField.maybeNegate(matcher!.matches(this.value(task)), filterMethod);
-        });
+        const negate = operator.match(/not/) !== null;
+        return FilterOrErrorMessage.fromFilter(this.getFilter(matcher, negate));
     }
 
     public static stringIncludesCaseInsensitive(haystack: string, needle: string): boolean {
         return SubstringMatcher.stringIncludesCaseInsensitive(haystack, needle);
     }
 
+    protected fieldPattern(): string {
```

Comment:
I know this is not public, but I think it is really fundamental, and would be worth adding JSDoc. Partly as it wasn't clear to me what was meant by pattern, and also it sounds like it might include slashes, but it doesn't...

Reply:
This has now JSDoc, too.

Reply:
Thank you for the documentation improvements.

I think this is another vocabulary thing, but I'm confused by `Returns a regexp pattern` in the docs.

I'm a bit confused as to why something called a regex pattern is returning something that does not begin and end with `/`.

But realistically, anyone adding a new class is going to be looking at the existing implementations for examples, and it doesn't take long to see that they don't add `/`.

So feel free to resolve this conversation!

Reply:
Thank you. I've made a note of that - at some point I intend to create one or two glossary files...
Some names focussed on users
And some on developers.

(We've a total mixture of filter, instruction, command and probably others, in the user docs)

Reply:
We could make the method defining the operator part of the filterRegExp return a RegExp as well (using slashes instead of quotes), but we actually only need the source of that RegExp and would then need to get it back with `.source`. Easier then to provide the source directly, no need to compile the parts.

Yes, documentation often is confusing because of inconsistent use of terminology, particularly when different people are working on it. A glossary can help, particularly for those writing code and documentation.

Reply:
> We could make the method defining the operator part of the filterRegExp return a RegExp as well (using slashes instead of quotes), but we actually only need the source of that RegExp and would then need to get it back with `.source`. Easier then to provide the source directly, no need to compile the parts.

That's all fine!

> Yes, documentation often is confusing because of inconsistent use of terminology, particularly when different people are working on it. A glossary can help, particularly for those writing code and documentation.

Totally agreed... I've only been responsible for Tasks for just under 4 months, and my backlog - including a backlog of things to log in GitHub is huge... 

But progress is definitely being made!